test/unit/postprocessing/test_threshold_optimization.py

```diff
@@ -320,144 +314,123 @@ def _average_prediction_for_label(label, sensitive_feature_value,
-@pytest.mark.parametrize("X_transform", ALLOWED_INPUT_DATA_TYPES)
-@pytest.mark.parametrize("y_transform", ALLOWED_INPUT_DATA_TYPES)
-@pytest.mark.parametrize("sensitive_features_transform", ALLOWED_INPUT_DATA_TYPES)
+    for a in data_X_y_sf.feature_names:
+        pos_indices = (mapped_sensitive_features == a) * (labels_ex == 1)
+        neg_indices = (mapped_sensitive_features == a) * (labels_ex == 0)
+        average_probs_positive_indices = np.average(predictions[pos_indices], axis=0)
+        average_probs_negative_indices = np.average(predictions[neg_indices], axis=0)
+        assert np.isclose(average_probs_positive_indices[0], expected_ps[_POS_P0])
+        assert np.isclose(average_probs_positive_indices[1], expected_ps[_POS_P1])
+        assert np.isclose(average_probs_negative_indices[0], expected_ps[_NEG_P0])
+        assert np.isclose(average_probs_negative_indices[1], expected_ps[_NEG_P1])
+
+
 @pytest.mark.parametrize("constraints", [DEMOGRAPHIC_PARITY, EQUALIZED_ODDS])
-def test_predict_output_0_or_1(sensitive_features, sensitive_feature_names, X_transform,
-                               y_transform, sensitive_features_transform, constraints):
-    X = X_transform(_format_as_list_of_lists(sensitive_features))
-    y = y_transform(labels_ex)
-    sensitive_features_ = sensitive_features_transform(sensitive_features)
+def test_predict_output_0_or_1(data_X_y_sf, constraints):
     adjusted_predictor = ThresholdOptimizer(unconstrained_predictor=ExamplePredictor(),
                                             constraints=constraints)
-    adjusted_predictor.fit(X, y, sensitive_features=sensitive_features_)
+    adjusted_predictor.fit(data_X_y_sf.X, data_X_y_sf.y,
+                           sensitive_features=data_X_y_sf.sensitive_features)
 
-    predictions = adjusted_predictor.predict(X, sensitive_features=sensitive_features_)
+    predictions = adjusted_predictor.predict(
+        data_X_y_sf.X, sensitive_features=data_X_y_sf.sensitive_features)
     for prediction in predictions:
         assert prediction in [0, 1]
 
 
-@pytest.mark.parametrize("sensitive_features,sensitive_feature_names",
-                         [(sensitive_features_ex1, sensitive_feature_names_ex1),
-                          (sensitive_features_ex2, sensitive_feature_names_ex2)])
-@pytest.mark.parametrize("X_transform", ALLOWED_INPUT_DATA_TYPES)
-@pytest.mark.parametrize("y_transform", ALLOWED_INPUT_DATA_TYPES)
 @pytest.mark.parametrize("constraints", [DEMOGRAPHIC_PARITY, EQUALIZED_ODDS])
-def test_predict_multiple_sensitive_features_columns_error(
-        sensitive_features, sensitive_feature_names, X_transform, y_transform, constraints):
-    X = X_transform(_format_as_list_of_lists(sensitive_features))
-    y = y_transform(labels_ex)
-    sensitive_features_ = pd.DataFrame({"A1": sensitive_features, "A2": sensitive_features})
+def test_predict_different_argument_lengths(data_X_y_sf, constraints):
```

Comment:
Hmmm looking and the 'old' vs 'new' test names, I'm note sure `diff` is being very helpful here :-(

Reply:
I don't think I understand. Can you elaborate?

Reply:
The two tests being matched together by `diff` are `test_predict_multiple_sensitive_features_columns_error` and `test_predict_different_argument_lengths` which are probably not testing remotely the same thing.... and hence looking at the differences is unlikely to be edifying. This is a complaint about `diff` and not the PR :-/

Reply:
Ah I see. Sometimes the side-by-side view helps, although copying out of it is a horrific experience :-(